autopilot/agent.go

```diff
@@ -520,11 +531,19 @@ func (a *Agent) controller() {
 					continue
 				}
 
+				// Skip candidates which we are already trying
```

Comment:
Small comment (although not touched in this PR) about the `MaxPendingOpens` check above this line: I think `pendingOpens` will not change inside the loop, as it is locked by `pendingMtx`. The check could be lifted up and performed before the loop starts?

Reply:
Good catch, should help reduce redundant log msgs as well!